src/client.ts

```diff
@@ -924,12 +926,13 @@ export class ConnectionImpl implements Executor {
     query: string,
     asJson: boolean,
     expectOne: boolean,
-    alwaysDescribe: boolean
+    alwaysDescribe: boolean,
+    options?: ParseOptions
   ): Promise<[number, ICodec, ICodec, Buffer | null, Buffer | null]> {
     const wb = new WriteMessageBuffer();
 
     wb.beginMessage(chars.$P)
-      .writeInt16(0) // no headers
+      .writeHeaders(options?.headers ?? {})
```

Comment:
```suggestion
      .writeHeaders(options?.headers ?? null)
```

and bail out on null in `writeHeaders` -- that will be a bit faster.